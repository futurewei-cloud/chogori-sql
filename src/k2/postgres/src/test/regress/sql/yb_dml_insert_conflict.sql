--
-- insert...on conflict on constraint
--
CREATE TABLE tab (i int PRIMARY KEY, j int CONSTRAINT tab_j_constraint UNIQUE);
CREATE UNIQUE INDEX ON tab (j);
CREATE UNIQUE INDEX tab_idx ON tab (j);
-- Insert some data
INSERT INTO tab VALUES (1, 1);
INSERT INTO tab VALUES (200, 200);
-- Use generated constraint name for primary key.
INSERT INTO tab VALUES (1, 7) ON CONFLICT ON CONSTRAINT tab_pkey DO UPDATE
    SET j = tab.j + excluded.j;
-- Use select to verify result.
SELECT * FROM tab;
-- Use system generated name for unique index.
INSERT INTO tab VALUES (1, 200) ON CONFLICT ON CONSTRAINT tab_j_idx DO UPDATE
    SET j = tab.j + excluded.j;
-- Error: Name of index is not a constraint name.
INSERT INTO tab VALUES (1, 1) ON CONFLICT ON CONSTRAINT tab_idx DO NOTHING;
-- Use conflict for unique column
INSERT INTO tab VALUES (1, 200) ON CONFLICT (j) DO UPDATE
    SET j = tab.j + excluded.j;
-- Use SELECT to verify result.
SELECT * FROM tab;
-- Use conflict for unique constraint - noop
INSERT INTO tab VALUES (1, 400) ON CONFLICT ON CONSTRAINT tab_j_constraint DO NOTHING;
-- Use SELECT to verify result.
SELECT * FROM tab;
-- Use conflict for unique constraint - update
INSERT INTO tab VALUES (1, 400) ON CONFLICT ON CONSTRAINT tab_j_constraint DO UPDATE
    SET j = tab.j + excluded.j;
-- Use SELECT to verify result.
SELECT * FROM tab;
