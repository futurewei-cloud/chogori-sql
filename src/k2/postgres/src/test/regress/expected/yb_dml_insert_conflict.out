--
-- insert...on conflict on constraint
--
CREATE TABLE tab (i int PRIMARY KEY, j int CONSTRAINT tab_j_constraint UNIQUE);
CREATE UNIQUE INDEX ON tab (j);
CREATE UNIQUE INDEX tab_idx ON tab (j);
-- Insert some data
INSERT INTO tab VALUES (1, 1);
INSERT INTO tab VALUES (200, 200);
-- Use generated constraint name for primary key.
INSERT INTO tab VALUES (1, 7) ON CONFLICT ON CONSTRAINT tab_pkey DO UPDATE
    SET j = tab.j + excluded.j;
-- Use select to verify result.
SELECT * FROM tab;
  i  |  j
-----+-----
   1 |   8
 200 | 200
(2 rows)

-- Use system generated name for unique index.
INSERT INTO tab VALUES (1, 200) ON CONFLICT ON CONSTRAINT tab_j_idx DO UPDATE
    SET j = tab.j + excluded.j;
ERROR:  constraint "tab_j_idx" for table "tab" does not exist
-- Error: Name of index is not a constraint name.
INSERT INTO tab VALUES (1, 1) ON CONFLICT ON CONSTRAINT tab_idx DO NOTHING;
ERROR:  constraint "tab_idx" for table "tab" does not exist
-- Use conflict for unique column
INSERT INTO tab VALUES (1, 200) ON CONFLICT (j) DO UPDATE
    SET j = tab.j + excluded.j;
-- Use SELECT to verify result.
SELECT * FROM tab;
  i  |  j
-----+-----
   1 |   8
 200 | 400
(2 rows)

-- Use conflict for unique constraint - noop
INSERT INTO tab VALUES (1, 400) ON CONFLICT ON CONSTRAINT tab_j_constraint DO NOTHING;
-- Use SELECT to verify result.
SELECT * FROM tab;
  i  |  j
-----+-----
   1 |   8
 200 | 400
(2 rows)

-- Use conflict for unique constraint - update
INSERT INTO tab VALUES (1, 400) ON CONFLICT ON CONSTRAINT tab_j_constraint DO UPDATE
    SET j = tab.j + excluded.j;
-- Use SELECT to verify result.
SELECT * FROM tab;
  i  |  j
-----+-----
   1 |   8
 200 | 800
(2 rows)
